### PR TITLE
Allow setting a custom OMR prefix (required for 1.3.0 cert. test cases)

### PR DESCRIPTION
**Pull request by suveshpratapa**
The latest 1.3.0 test spec includes cases where the harness is asked to set a manual OMR prefix on a BR reference device in order to test some OMR selection criteria. This change expands the `br omrprefix` command to take an optional prefix value to set our own OMR prefix.

### CODE REVIEW

**Comment by suveshpratapa** [comment]
@jwhui @simonlingoogle @edmont

File: src/core/border_router/routing_manager.cpp [hunk 1]
```diff
@@ -152,6 +152,18 @@ Error RoutingManager::GetOmrPrefix(Ip6::Prefix &aPrefix)
     return error;
 }
 
+Error RoutingManager::SetOmrPrefix(Ip6::Prefix &aPrefix)
+{
+    Error error = kErrorNone;
+
+    VerifyOrExit(IsInitialized(), error = kErrorInvalidState);
+    mLocalOmrPrefix = aPrefix;
+    SuccessOrExit(PublishLocalOmrPrefix());
```
Comment (abtink) [inline src/core/border_router/routing_manager.cpp:161]: I am not sure about the call to `PublishLocalOmrPrefix()` here. 

This assumes that OMR prefix should be added in network data (which may not be the case, e.g. there may already be other OMR prefixes in netdata which are favored over this BR's local OMR prefix.

The decision whether local OMR prefix needs to be added or removed to Network Data is done from `EvaluateOmrPrefix()`.

Another thing to consider:
- if the previous `mLocalOmrPrefix` was added in network data, we need to remove it first, then change the value to the new given prefix and decide if we should add the new one in netdata.

**Review by abtink — COMMENTED** [review]
Thanks @suveshpratapa.

> The latest 1.3.0 test spec includes cases where the harness is asked to set a manual OMR prefix on a BR reference device in order to test some OMR selection criteria. This change expands the br omrprefix command to take an optional prefix value to set our own OMR prefix.

Would it work if we instead  directly add OMR prefix in network data ?
- Using `otBorderRouterAddOnMeshPrefix()` API
- And/or it related CLI command `prefix add`

This way we don't need to go through `RoutingManager` and add new API.

**Review by edmont — COMMENTED** [review]
Shouldn't we wrap this functionality around `OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE`?

Also please make sure to increase `OPENTHREAD_API_VERSION` and run `./script/make-pretty`.

File: src/core/border_router/routing_manager.cpp (hunk 1 above)
Comment (abtink) [reply src/core/border_router/routing_manager.cpp:161]: `br omrprefix` returns the local OMR prefix (by `RoutingManager` on BR) which would be added to netdata (if there is not an existing and more favored omr prefix in net data).

But practically OMR prefix is a prefix in netdata which meets certain rules: 64 bit length with set of flags (slaac, on-mesh, preferred , etc). So we can add it directly in netdata.

Can you please share more details about the test case that requires specific OMR prefix?

If the test is to see behavior of BR when there is some specific OMR prefix in netdata we can add it directly in netdata.

Comment (suveshpratapa) [reply src/core/border_router/routing_manager.cpp:161]: See "Thread 1.3.0 test spec TC 1.8. Reachability - Multiple BRs - Single Thread / Single IPv6 Infrastructure - OMR prefix selection"

On a testbed device BR_2 in that test, the initial step is:
```
Harness configures BR_2 with a chosen OMR prefix, OMR_1.
```

Additionally, further steps have:
```
Harness instructs BR_2 to add an OMR prefix OMR_3 numerically lower than OMR_2, but with same preference “Low”.
```

```
Harness instructs BR_2 to add new OMR prefix (OMR_4) numerically higher than OMR_2, but with same preference “Low”.
```

```
Harness instructs BR_2 to Update the numerically higher OMR_4  with a higher preference value 00 (‘Medium’)
```

Comment (abtink) [reply src/core/border_router/routing_manager.cpp:161]: Thanks @suveshpratapa.

I checked the test-case. I think we do not need an API on `RoutingManager` to set OMR prefix, and actually would be better to add/remove prefixes directly in netdata.

The local OMR prefix from the `br omrprfix` is managed by `RoutingManager` and it may not be added in network data in all situations (`RoutingManager` monitors entries in net data to see its local prefix is favored over other entries and then decide to add/remove it in net data). Also it is always added as `Low` preference in net data (which I guess is required by spec for self-generated OMR prefix).

-----

The test-case is covering behavior of DUT when other BRs add/remove different OMR prefixes in network data. (basically testing the `RoutingManager` behavior on DUT).

We can directly add the entries in network data from other BRs and we don't need to use the `RoutingManager` module on other BRs. The test mentions:

```
Harness configures BR_2 with a chosen OMR prefix, OMR_1. 
Note: The default automatic OMR prefix generation is disabled on BR_2 for the duration of this test.
```

Basically we should disable `RoutingManager` on `BR_1` (not do `br enable`) and then can directly add/remove prefixes in netdata (using `prefix add` and `prefix remove` which let us set the prf as well.

Comment (abtink) [reply src/core/border_router/routing_manager.cpp:161]: > From what you're saying here, we can do the same thing for selecting a different OMR prefix, by basically doing `br disable` followed by `prefix add`. Is that right?

Yes. Exactaly. This should work.
 
One point, after `prefix add` or `prefix remove` we need `netdata register` to ask the changes to be registered with leader.

**Comment by suveshpratapa** [comment]
Closing as this is not required.
